odoo/addons/base/res/res_country.py

```diff
@@ -105,7 +105,7 @@ class CountryState(models.Model):
     name = fields.Char(string='State Name', required=True,
                help='Administrative divisions of a country. E.g. Fed. State, Departement, Canton')
     code = fields.Char(string='State Code', help='The state code.', required=True)
-
+    image = fields.Binary(attachment=True, required=False)
```

Comment:
if it is not used anywhere else than in the form view of the state, it doesn't make much sense to add it, definitely something for a custom module